src/main/java/com/_up/megastore/data/repositories/IProductRepository.java

```diff
@@ -10,4 +10,5 @@
 public interface IProductRepository extends JpaRepository<Product, UUID> {
     boolean existsByProductIdAndDeletedTrue(UUID productId);
     Page<Product> findProductsByDeletedIsFalseAndNameContainingIgnoreCase(String name, Pageable pageable);
+    boolean existsByNameAndDeletedIsFalse(String name);
```

Comment:
Agregaria el IgnoreCase para que salte error si se quiera guardar "REMERA" y ya existe "Remera"

```suggestion
    boolean existsByNameIgnoreCaseAndDeletedIsFalse(String name);
```